### PR TITLE
Feature/047 api documentation

### PR DESCRIPTION
**Pull request by gopika-na**
Ready for review

### CODE REVIEW

**Comment by gopika-na** [comment]
@kristinpeterson , gentle reminder for review. Thank you.

File: lib/collections.js [hunk 1]
```diff
@@ -5,18 +5,23 @@
...
 var helpers = require("./helpers");
 var Collections;
-
-// Wrapper for AT&T M2X Collection API
-//
-// https://m2x.att.com/developer/documentation/collections
+/**
+ * @module Collections
+ * @description Method for [Wrapper for AT&T M2X Collections API]{@link https://m2x.att.com/developer/documentation/v2/collections} endpoint.
+ * @param client {str}
```
Comment (mentos) [inline lib/collections.js]: I believe this should be `@param client {object}`.

File: lib/commands.js [hunk 2]
```diff
@@ -4,15 +4,23 @@ var helpers = require("./helpers");
 
 var Commands;
 
-// Wrapper for AT&T M2X Commands API
-// https://m2x.att.com/developer/documentation/v2/commands
+/**
+ * @module Commands
+ * @description Method for [Wrapper for AT&T M2X Commands API]{@link https://m2x.att.com/developer/documentation/v2/commands} endpoint.
+ * @param client {str}
```
Comment (mentos) [inline lib/commands.js]: I believe this should be `@param client {object}`.

File: lib/distributions.js [hunk 3]
```diff
@@ -3,16 +3,23 @@
 var helpers = require("./helpers");
 var Distributions;
 
-// Wrapper for AT&T M2X Distribution API
-//
-// https://m2x.att.com/developer/documentation/distribution
+/**
+ * @module Distributions
+ * @description Method for [Wrapper for AT&T M2X Distribution API]{@link https://m2x.att.com/developer/documentation/distribution} endpoint.
+ * @param client {str}
```
Comment (mentos) [inline lib/distributions.js]: I believe this should be `@param client {object}`.

File: lib/jobs.js [hunk 4]
```diff
@@ -3,17 +3,25 @@
 var helpers = require("./helpers");
 var Jobs;
 
-// Wrapper for AT&T M2X Jobs API
-//
-// https://m2x.att.com/developer/documentation/v2/jobs
+/**
+ * @module Jobs
+ * @description Method for [Wrapper for AT&T M2X Jobs API]{@link https://m2x.att.com/developer/documentation/v2/jobs} endpoint.
+ * @param client {str}
```
Comment (mentos) [inline lib/jobs.js]: I believe this should be `@param client {object}`.

File: lib/keys.js [hunk 5]
```diff
@@ -2,61 +2,85 @@
 
 var helpers = require("./helpers");
 
-// Wrapper for AT&T M2X Keys API
-//
-// https://m2x.att.com/developer/documentation/keys
+/**
+ * @module Keys
+ * @description Method for [Wrapper for AT&T M2X Keys API]{@link https://m2x.att.com/developer/documentation/keys} endpoint.
+ * @param client {str}
```
Comment (mentos) [inline lib/keys.js]: I believe this should be `@param client {object}`.

File: lib/collections.js (hunk 1 above)
Comment (mentos) [inline lib/collections.js]: I believe this should be `@param client {object}`.

File: lib/collections.js [hunk 6]
```diff
@@ -25,72 +30,106 @@ Collections.prototype.list = function(params, callback) {
   return this.client.get("/collections", { qs: params || {} }, callback);
 };
 
-// Create a new collection
-//
-// https://m2x.att.com/developer/documentation/v2/collections#Create-Collection
+/**
+ * @memberOf Collections
+ * @description Method for [Create Collection]{@link https://m2x.att.com/developer/documentation/v2/collections#Create-Collection} endpoint.
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/collections.js]: I believe this should be `@param params {object}`.

File: lib/collections.js [hunk 7]
```diff
@@ -25,72 +30,106 @@ Collections.prototype.list = function(params, callback) {
...
 };
 
-// Update a collection
-//
-// https://m2x.att.com/developer/documentation/v2/collections#Update-Collection-Details
+/**
+ * @memberOf Collections
+ * @description Method for [Update Collection Details]{@link https://m2x.att.com/developer/documentation/v2/collections#Update-Collection-Details} endpoint.
+ * @param id ID of the Collection to update
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/collections.js]: I believe this should be `@param params {object}`.

File: lib/collections.js [hunk 8]
```diff
@@ -25,72 +30,106 @@ Collections.prototype.list = function(params, callback) {
...
 };
 
-// Update Collection Metadata
-//
-// https://m2x.att.com/developer/documentation/v2/collections#Update-Collection-Metadata
+/**
+ * @memberOf Collections
+ * @description Method for [Update Collection Metadata]{@link https://m2x.att.com/developer/documentation/v2/collections#Update-Collection-Metadata} endpoint.
+ * @param id {str} ID of the Collection to update metadata
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/collections.js]: I believe this should be `@param params {object}`.

File: lib/collections.js [hunk 9]
```diff
@@ -5,18 +5,23 @@
...
 };
 
-// Retrieve the list of collections accessible by the authenticated API key that
-// meet the search criteria
-//
-// https://m2x.att.com/developer/documentation/v2/collections#List-collections
+/**
+ * @memberOf Collections
+ * @description Method for [List Collections]{@link https://m2x.att.com/developer/documentation/v2/collections#List-collections} endpoint.
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/collections.js]: I believe this should be `@param params {object}`.

File: lib/commands.js [hunk 10]
```diff
@@ -4,15 +4,23 @@ var helpers = require("./helpers");
...
     this.client = client;
 };
 
-// List Sent Commands
-//
-// https://m2x.att.com/developer/documentation/v2/commands#List-Sent-Commands
+/**
+ * @memberOf Commands
+ * @description Method for [List Sent Commands]{@link https://m2x.att.com/developer/documentation/v2/commands#List-Sent-Commands} endpoint.
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/commands.js]: I believe this should be `@param params {object}`.

File: lib/commands.js [hunk 11]
```diff
@@ -21,19 +29,24 @@ Commands.prototype.list = function(params, callback) {
...
-// Send Command
-// Send a command with the given name to the given target devices.
-// The name should be a custom string defined by the user and understood by
-// the device.
-//
-// https://m2x.att.com/developer/documentation/v2/commands#Send-Command
+/**
+ * @memberOf Commands
+ * @description Method for [Send Command]{@link https://m2x.att.com/developer/documentation/v2/commands#Send-Command} endpoint.
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/commands.js]: I believe this should be `@param params {object}`.

File: lib/distributions.js [hunk 12]
```diff
@@ -3,16 +3,23 @@
...
     this.client = client;
 };
 
-// Retrieve a list of device distributions
-//
-// https://m2x.att.com/developer/documentation/v2/distribution#List-Distributions
+/**
+ * @memberOf Distributions
+ * @description Method for [List Distributions]{@link https://m2x.att.com/developer/documentation/v2/distribution#List-Distributions} endpoint.
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/distributions.js]: I believe this should be `@param params {object}`.

File: lib/distributions.js [hunk 13]
```diff
@@ -21,23 +28,36 @@ Distributions.prototype.list = function(params, callback) {
     return this.client.get("/distributions", { qs: params || {} }, callback);
 };
 
-// Create a new device distribution
-//
-// https://m2x.att.com/developer/documentation/v2/distribution#Create-Distribution
+/**
+ * @memberOf Distributions
+ * @description Method for [Create Distribution]{@link https://m2x.att.com/developer/documentation/v2/distribution#Create-Distribution} endpoint.
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/distributions.js]: I believe this should be `@param params {object}`.

File: lib/distributions.js [hunk 14]
```diff
@@ -21,23 +28,36 @@ Distributions.prototype.list = function(params, callback) {
...
 };
 
-// Update an existing device distribution
-//
-// https://m2x.att.com/developer/documentation/v2/distribution#Update-Distribution-Details
+/**
+ * @memberOf Distributions
+ * @description Method for [Update Distribution Details]{@link https://m2x.att.com/developer/documentation/v2/distribution#Update-Distribution-Details} endpoint.
+ * @param id {str} ID of the Distribution to update
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/distributions.js]: I believe this should be `@param params {object}`.

File: lib/distributions.js [hunk 15]
```diff
@@ -46,46 +66,69 @@ Distributions.prototype.update = function(id, params, callback) {
...
 };
 
-// Add a new device to an existing device distribution
-//
-// https://m2x.att.com/developer/documentation/v2/distribution#Add-Device-to-an-existing-Distribution
+/**
+ * @memberOf Distributions
+ * @description Method for [Add Device to an existing Distribution]{@link https://m2x.att.com/developer/documentation/v2/distribution#Add-Device-to-an-existing-Distribution} endpoint.
+ * @param id {str} ID of the Distribution to add device
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/distributions.js]: I believe this should be `@param params {object}`.

File: lib/distributions.js [hunk 16]
```diff
@@ -46,46 +66,69 @@ Distributions.prototype.update = function(id, params, callback) {
...
 
-// Create/Update a data stream associated with the distribution
-//
-// https://m2x.att.com/developer/documentation/v2/distribution#Create-Update-Data-Stream
+/**
+ * @memberOf Distributions
+ * @description Method for [Create/Update Data Stream]{@link https://m2x.att.com/developer/documentation/v2/distribution#Create-Update-Data-Stream} endpoint.
+ * @param id {str} ID of the Distribution to create/update data streams
+ * @param name {str} Name of the Data stream
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/distributions.js]: I believe this should be `@param params {object}`.

File: lib/distributions.js [hunk 17]
```diff
@@ -97,47 +140,67 @@ Distributions.prototype.updateDataStream = function(id, name, params, callback)
...
 };
 
-// Update Distribution Metadata
-//
-// https://m2x.att.com/developer/documentation/v2/distribution#Update-Distribution-Metadata
+/**
+ * @memberOf Distributions
+ * @description Method for [Update Distribution Metadata]{@link https://m2x.att.com/developer/documentation/v2/distribution#Update-Distribution-Metadata} endpoint.
+ * @param id {str} ID of the Distribution to update metadata
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/distributions.js]: I believe this should be `@param params {object}`.

File: lib/keys.js [hunk 18]
```diff
@@ -2,61 +2,85 @@
...
     return this.client.del(helpers.url("/keys/%s", key), callback);
 };
 
-// Create a new API Key
-//
-// https://m2x.att.com/developer/documentation/v2/keys#Create-Key
+/**
+ * @memberOf Keys
+ * @description Method for [Create Key]{@link https://m2x.att.com/developer/documentation/v2/keys#Create-Key} endpoint.
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/keys.js]: I believe this should be `@param params {object}`.

File: lib/keys.js [hunk 19]
```diff
@@ -2,61 +2,85 @@
...
 };
 
-// Update API Key properties
-//
-// https://m2x.att.com/developer/documentation/v2/keys#Update-Key
+/**
+ * @memberOf Keys
+ * @description Method for [Update Key]{@link https://m2x.att.com/developer/documentation/v2/keys#Update-Key} endpoint.
+ * @param key {str} Key associated with a developer account
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/keys.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 20]
```diff
@@ -3,22 +3,26 @@
 var helpers = require("./helpers");
 var Devices;
 
-// Wrapper for AT&T M2X Device API
-//
-// https://m2x.att.com/developer/documentation/device
+/**
+ * @module Devices
+ * @description Method for [Wrapper for AT&T M2X Device API]{@link https://m2x.att.com/developer/documentation/device} endpoint.
+ * @param client {str}
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param client {object}`.

File: lib/devices.js [hunk 21]
```diff
@@ -3,22 +3,26 @@
...
-// that have been marked as public, allowing them to read public Device
-// metadata, locations, streams list, and view each Devices' stream metadata
-// and its values.
-//
-// https://m2x.att.com/developer/documentation/v2/device#List-Search-Public-Devices-Catalog
+/**
+ * @memberOf Devices
+ * @description Method for [List/search Public Devices Catalog]{@link https://m2x.att.com/developer/documentation/v2/device#List-Public-Devices-Catalog} endpoint.
+ * This allows unauthenticated users to search Devices from other users that have been marked as public, allowing them to read public Device metadata, locations, streams list, and view each Devices' stream metadata and its values.
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 22]
```diff
@@ -27,9 +31,13 @@ Devices.prototype.catalog = function(params, callback) {
     return this.client.get("/devices/catalog", { qs: params || {} }, callback);
 };
 
-// Retrieve the list of devices accessible by the authenticated API key
-//
-// https://m2x.att.com/developer/documentation/v2/device#List-Devices
+/**
+ * @memberOf Devices
+ * @description Method for [List Devices]{@link https://m2x.att.com/developer/documentation/v2/device#List-Devices} endpoint.
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 23]
```diff
@@ -38,10 +46,13 @@ Devices.prototype.list = function(params, callback) {
...
 };
 
-// Retrieve the list of devices accessible by the authenticated API key that
-// meet the search criteria.
-//
-// https://m2x.att.com/developer/documentation/v2/device#Search-Devices
+/**
+ * @memberOf Devices
+ * @description Method for [Search Devices]{@link https://m2x.att.com/developer/documentation/v2/device#Search-Devices} endpoint.
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 24]
```diff
@@ -53,62 +64,88 @@ Devices.prototype.search = function(params, callback) {
...
     return this.client.get("/devices/tags", callback);
 };
 
-// Create a new device
-//
-// https://m2x.att.com/developer/documentation/v2/device#Create-Device
+/**
+ * @memberOf Devices
+ * @description Method for [Create Device]{@link https://m2x.att.com/developer/documentation/v2/device#Create-Device} endpoint.
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 25]
```diff
@@ -53,62 +64,88 @@ Devices.prototype.search = function(params, callback) {
...
 };
 
-// Update a device
-//
-// https://m2x.att.com/developer/documentation/v2/device#Update-Device-Details
+/**
+ * @memberOf Devices
+ * @description Method for [Update Device Details]{@link https://m2x.att.com/developer/documentation/v2/device#Update-Device-Details} endpoint.
+ * @param id {str} ID of the Device to update
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 26]
```diff
@@ -53,62 +64,88 @@ Devices.prototype.search = function(params, callback) {
...
-// Read Device Location History
-//
-// Get location history details of an existing Device.
-//
-// https://m2x.att.com/developer/documentation/v2/device#Read-Device-Location-History
+/**
+ * @memberOf Devices
+ * @description Method for [Read Device Location History]{@link https://m2x.att.com/developer/documentation/v2/device#Read-Device-Location-History} endpoint.
+ * @param id {str}  ID of the Device to retrieve location history
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 27]
```diff
@@ -53,62 +64,88 @@ Devices.prototype.search = function(params, callback) {
...
 };
 
-// Update the current location of the device
-//
-// https://m2x.att.com/developer/documentation/v2/device#Update-Device-Location
+/**
+ * @memberOf Devices
+ * @description Method for [Update Device Location]{@link https://m2x.att.com/developer/documentation/v2/device#Update-Device-Location} endpoint.
+ * @param id {str} ID of the Device to update location details
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 28]
```diff
@@ -117,11 +154,14 @@ Devices.prototype.updateLocation = function(id, params, callback) {
...
-// Delete Device Location History
-//
-// Delete location history detials of an existing Device
-//
-// https://m2x.att.com/developer/documentation/v2/device#Delete-Location-History
+/**
+ * @memberOf Devices
+ * @description Method for [Delete Location History]{@link https://m2x.att.com/developer/documentation/v2/device#Delete-Location-History} endpoint.
+ * @param id {str} ID of the Device to remove location history
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 29]
```diff
@@ -130,27 +170,37 @@ Devices.prototype.deleteLocationHistory = function(id, params, callback) {
...
 };
 
-// Update Device Metadata
-//
-// https://m2x.att.com/developer/documentation/v2/device#Update-Device-Metadata
+/**
+ * @memberOf Devices
+ * @description Method for [Update Device Metadata]{@link https://m2x.att.com/developer/documentation/v2/device#Update-Device-Metadata} endpoint.
+ * @param id {str} ID of the Device to update device metadata
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 30]
```diff
@@ -170,18 +226,27 @@ Devices.prototype.updateMetadataField = function(id, field, value, callback) {
...
-// If the stream doesn't exist it will create it.
-//
-// https://m2x.att.com/developer/documentation/v2/device#Create-Update-Data-Stream
+/**
+ * @memberOf Devices
+ * @description Method for [Create Update Data Stream]{@link https://m2x.att.com/developer/documentation/v2/device#Create-Update-Data-Stream} endpoint.
+ * If the stream doesn't exist it will create it.
+ * @param id {str} ID of the Device to update stream's properties
+ * @param name {str} Name of the stream to be updated
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 31]
```diff
@@ -195,31 +260,16 @@ Devices.prototype.updateStream = function(id, name, params, callback) {
...
-//
-// For properties details, see:
-// https://m2x.att.com/developer/documentation/v2/device#Create-Update-Data-Stream
+/**
+ * @memberOf Devices
+ * @description Method for [Create Update Data Stream]{@link https://m2x.att.com/developer/documentation/v2/device#Create-Update-Data-Stream} endpoint.
+ * If the stream doesn't exist it will create it.
+ * @param id {str} ID of the Device to update stream's properties
+ * @param name {Array} Name of the streams to be updated
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 32]
```diff
@@ -244,9 +294,15 @@ Devices.prototype.updateStreams = function(id, names, params, callback) {
...
 
-// Set the stream value
-//
-// https://m2x.att.com/developer/documentation/v2/device#Update-Data-Stream-Value
+/**
+ * @memberOf Devices
+ * @description Method for [Update Data Stream Value]{@link https://m2x.att.com/developer/documentation/v2/device#Update-Data-Stream-Value} endpoint.
+ * @param id {str} ID of the Device to set the stream value
+ * @param name {str} Name of the stream to be set
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 33]
```diff
@@ -255,21 +311,30 @@ Devices.prototype.setStreamValue = function(id, name, params, callback) {
...
-// specific device, sorted in reverse chronological order (most
-// recent values first).
-//
-// https://m2x.att.com/developer/documentation/v2/device#List-Data-Stream-Values
+/**
+ * @memberOf Devices
+ * @description Method for [List Data Stream Values]{@link https://m2x.att.com/developer/documentation/v2/device#List-Data-Stream-Values} endpoint.
+ * @param id {str} ID of the Device
+ * @param name {str} Name of the stream to retrieve
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 34]
```diff
@@ -281,9 +346,14 @@ Devices.prototype.streamValues = function(id, name, params, callback) {
...
 };
 
-// List values from all data stream associated with a specific device
-//
-// https://m2x.att.com/developer/documentation/v2/device#List-Values-from-all-Data-Streams-of-a-Device
+/**
+ * @memberOf Devices
+ * @description Method for [List Values from all Data Streams of a Device]{@link https://m2x.att.com/developer/documentation/v2/device#List-Values-from-all-Data-Streams-of-a-Device} endpoint.
+ * @param id {str} ID of the Device
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 35]
```diff
@@ -295,9 +365,14 @@ Devices.prototype.values = function(id, params, callback) {
...
 };
 
-// Export Values from all Data Streams of a Device
-//
-// https://m2x.att.com/developer/documentation/v2/device#Export-Values-from-all-Data-Streams-of-a-Device
+/**
+ * @memberOf Devices
+ * @description Method for [Export Values from all Data Streams of a Device]{@link https://m2x.att.com/developer/documentation/v2/device#Export-Values-from-all-Data-Streams-of-a-Device} endpoint.
+ * @param id {str} ID of the Device
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 36]
```diff
@@ -309,9 +384,14 @@ Devices.prototype.valuesExport = function(id, params, callback) {
...
 };
 
-// Search Values from all Data Streams of a Device
-//
-// https://m2x.att.com/developer/documentation/v2/device#Search-Values-from-all-Data-Streams-of-a-Device
+/**
+ * @memberOf Devices
+ * @description Method for [Search Values from all Data Streams of a Device]{@link https://m2x.att.com/developer/documentation/v2/device#Search-Values-from-all-Data-Streams-of-a-Device} endpoint.
+ * @param id {str} ID of the Device
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 37]
```diff
@@ -323,9 +403,15 @@ Devices.prototype.valuesSearch = function(id, params, callback) {
...
 
-// Sample values from an existing stream
-//
-// https://m2x.att.com/developer/documentation/v2/device#Data-Stream-Sampling
+/**
+ * @memberOf Devices
+ * @description Method for [Data Stream Sampling]{@link https://m2x.att.com/developer/documentation/v2/device#Data-Stream-Sampling} endpoint.
+ * @param id {str} ID of the Device
+ * @param name {str} Name of the stream to retrieve
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 38]
```diff
@@ -338,9 +424,15 @@ Devices.prototype.sampleStreamValues = function(id, name, params, callback) {
...
 
-// Return the stream stats
-//
-// https://m2x.att.com/developer/documentation/v2/device#Data-Stream-Stats
+/**
+ * @memberOf Devices
+ * @description Method for [Data Stream Stats]{@link https://m2x.att.com/developer/documentation/v2/device#Data-Stream-Stats} endpoint.
+ * @param id {str} ID of the Device
+ * @param name {str} Name of the stream to retrieve
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 39]
```diff
@@ -364,9 +462,16 @@ Devices.prototype.postValues = function(id, name, values, callback) {
...
-// Delete values from a stream by a date range
-//
-// https://m2x.att.com/developer/documentation/v2/device#Delete-Data-Stream-Values
+/**
+ *
+ * @memberOf Devices
+ * @description Method for [Delete Data Stream Values]{@link https://m2x.att.com/developer/documentation/v2/device#Delete-Data-Stream-Values} endpoint.
+ * @param id {str} ID of the Device
+ * @param name {str} Name of the existing stream to delete values
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 40]
```diff
@@ -375,93 +480,67 @@ Devices.prototype.deleteStreamValues = function(id, name, params, callback) {
...
-//
-// The optional `location` parameter can contain location information that will
-// be used to update the current location of the specified device
-//
-// https://m2x.att.com/developer/documentation/v2/device#Post-Device-Updates--Multiple-Values-to-Multiple-Streams-
+/**
+ * @memberOf Devices
+ * @description Method for [Post Device Updates(Multiple Values to Multiple Streams)]{@link https://m2x.att.com/developer/documentation/v2/device#Post-Device-Updates--Multiple-Values-to-Multiple-Streams-} endpoint.
+ * @param id {str} ID of the Device
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 41]
```diff
@@ -375,93 +480,67 @@ Devices.prototype.deleteStreamValues = function(id, name, params, callback) {
...
-//              elevation: 5
-//         }
-//      }
-//
-// https://m2x.att.com/developer/documentation/v2/device#Post-Device-Update--Single-Values-to-Multiple-Streams-
+/**
+ * @memberOf Devices
+ * @description Method for [Post Device Update(Single Value to Multiple Streams)]{@link https://m2x.att.com/developer/documentation/v2/device#Post-Device-Update--Single-Values-to-Multiple-Streams-} endpoint.
+ * @param id {str} ID of the Device
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 42]
```diff
@@ -375,93 +480,67 @@ Devices.prototype.deleteStreamValues = function(id, name, params, callback) {
...
-// Device's List of Received Commands
-//
-// Retrieve the list of recent commands sent to the current device (as given by the API key).
-//
-// https://m2x.att.com/developer/documentation/v2/commands#Device-s-List-of-Received-Commands
+/**
+ * @memberOf Devices
+ * @description Method for [Device's List of Received Commands]{@link https://m2x.att.com/developer/documentation/v2/commands#Device-s-List-of-Received-Commands} endpoint.
+ * @param id {str} ID of the Device to get list of received commands
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 43]
```diff
@@ -471,21 +550,29 @@ Devices.prototype.commands = function(id, params, callback) {
...
 
-// Process a command
-//
-// https://m2x.att.com/developer/documentation/v2/commands#Device-Marks-a-Command-as-Processed
+/**
+ * @memberOf Devices
+ * @description Method for [Device Marks a Command as Processed]{@link https://m2x.att.com/developer/documentation/v2/commands#Device-Marks-a-Command-as-Processed} endpoint.
+ * @param id {str} ID of the Device
+ * @param commandId {str} ID of the Command to retrieve
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 44]
```diff
@@ -499,9 +586,15 @@ Devices.prototype.processCommand = function(id, commandId, params, callback) {
...
 
-// Reject a command
-//
-// https://m2x.att.com/developer/documentation/v2/commands#Device-Marks-a-Command-as-Rejected
+/**
+ * @memberOf Devices
+ * @description Method for [Device Marks a Command as Rejected]{@link https://m2x.att.com/developer/documentation/v2/commands#Device-Marks-a-Command-as-Rejected} endpoint.
+ * @param id {str} ID of the Device
+ * @param commandId {str} ID of the Command to retrieve
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 45]
```diff
@@ -515,27 +608,48 @@ Devices.prototype.rejectCommand = function(id, commandId, params, callback) {
...
-// Creates a new API key associated to the device
-//
-// If a parameter named `stream` is supplied with a stream name, it
-// will create an API key associated with that stream only.
+/**
+ * @memberOf Devices
+ * @description Method for [Create Key]{@link https://m2x.att.com/developer/documentation/v2/keys#Create-Key} endpoint.
+ * If a parameter named `stream` is supplied with a stream name, it will create an API key associated with that stream only.
+ * @param id {str} ID of the Device to creates a new API key
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

File: lib/devices.js [hunk 46]
```diff
@@ -515,27 +608,48 @@ Devices.prototype.rejectCommand = function(id, commandId, params, callback) {
...
     this.keysAPI.create(helpers.extend(params, { device: id }), callback);
 };
 
-// Updates an API key properties
+/**
+ * @memberOf Devices
+ * @description Method for [Update Key]{@link https://m2x.att.com/developer/documentation/v2/keys#Update-Key} endpoint.
+ * @param id {str} ID of the Device to update API key
+ * @param key {str} API key to be updated
+ * @param params {params} Query parameters passed as keyword arguments. View M2X API Docs for listing of available parameters.
```
Comment (mentos) [inline lib/devices.js]: I believe this should be `@param params {object}`.

**Comment by gopika-na** [comment]
Ready for review.

**Pull request merged**